Push notification is not working on Android 8

```
07-24 17:03:48.441 2823-2823/? E/AndroidRuntime: FATAL EXCEPTION: main
	Process: chat.rocket.reactnative.debug, PID: 2823
	java.lang.RuntimeException: Unable to start receiver com.google.android.gms.gcm.GcmReceiver: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x1000010 pkg=chat.rocket.reactnative.debug cmp=chat.rocket.reactnative.debug/com.wix.reactnativenotifications.gcm.GcmMessageHandlerService (has extras) }: app is in background uid UidRecord{982ece1 u0a86 RCVR idle procs:1 seq(0,0,0)}
		at android.app.ActivityThread.handleReceiver(ActivityThread.java:3259)
		at android.app.ActivityThread.-wrap17(Unknown Source:0)
		at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1677)
		at android.os.Handler.dispatchMessage(Handler.java:105)
		at android.os.Looper.loop(Looper.java:164)
		at android.app.ActivityThread.main(ActivityThread.java:6541)
		at java.lang.reflect.Method.invoke(Native Method)
		at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
	Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.RECEIVE flg=0x1000010 pkg=chat.rocket.reactnative.debug cmp=chat.rocket.reactnative.debug/com.wix.reactnativenotifications.gcm.GcmMessageHandlerService (has extras) }: app is in background uid UidRecord{982ece1 u0a86 RCVR idle procs:1 seq(0,0,0)}
		at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1505)
		at android.app.ContextImpl.startService(ContextImpl.java:1461)
		at android.content.ContextWrapper.startService(ContextWrapper.java:644)
		at android.content.ContextWrapper.startService(ContextWrapper.java:644)
		at android.support.v4.content.WakefulBroadcastReceiver.startWakefulService(WakefulBroadcastReceiver.java:99)
		at com.google.android.gms.gcm.GcmReceiver.doStartService(Unknown Source:22)
		at com.google.android.gms.gcm.GcmReceiver.zzd(Unknown Source:0)
		at com.google.android.gms.gcm.GcmReceiver.onReceive(Unknown Source:98)
		at android.app.ActivityThread.handleReceiver(ActivityThread.java:3252)
```

--- a/android/app/src/main/java/chat/rocket/reactnative/CustomPushNotification.java
+++ b/android/app/src/main/java/chat/rocket/reactnative/CustomPushNotification.java
@@ -1,6 +1,8 @@
 package chat.rocket.reactnative;
 
 import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
 import android.app.PendingIntent;
 import android.content.Context;
 import android.content.res.Resources;
@@ -8,6 +10,7 @@ import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
 import android.media.AudioManager;
 import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
 import android.provider.Settings.System;
 import android.media.RingtoneManager;
@@ -33,8 +36,10 @@ public class CustomPushNotification extends PushNotification {
         String title = bundle.getString("title");
         String message = bundle.getString("message");
 
-        final Notification.Builder notification = new Notification.Builder(mContext);
-        notification
+        String CHANNEL_ID = "rocketchatrn_channel_01";
+        String CHANNEL_NAME = "All";
+
+        final Notification.Builder notification = new Notification.Builder(mContext)
             .setSmallIcon(smallIconResId)
             .setContentIntent(intent)
             .setContentTitle(title)
@@ -44,6 +49,17 @@ public class CustomPushNotification extends PushNotification {
             .setColor(mContext.getColor(R.color.notification_text))
             .setDefaults(Notification.DEFAULT_ALL)
             .setAutoCancel(true);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID,
+                                                                  CHANNEL_NAME,
+                                                                  NotificationManager.IMPORTANCE_DEFAULT);
+
+            final NotificationManager notificationManager = (NotificationManager) mContext.getSystemService(Context.NOTIFICATION_SERVICE);
+            notificationManager.createNotificationChannel(channel);
+
+            notification.setChannelId(CHANNEL_ID);
+        }
 
         Bitmap largeIconBitmap = BitmapFactory.decodeResource(res, largeIconResId);
         notification.setLargeIcon(largeIconBitmap);

--- a/package.json
+++ b/package.json
@@ -54,7 +54,7 @@
     "react-native-meteor": "^1.4.0",
     "react-native-modal": "^6.4.0",
     "react-native-navigation": "^1.1.474",
-    "react-native-notifications": "^1.1.19",
+    "react-native-notifications": "git+https://github.com/RocketChat/react-native-notifications.git",
     "react-native-optimized-flatlist": "^1.0.4",
     "react-native-picker-select": "^4.0.0",
     "react-native-responsive-ui": "^1.1.1",

